src/persistence/serialize.cpp

```diff
@@ -25,200 +25,66 @@ Most of functions in this file are unsafe unless otherwise specified.
 @warning Do not use them on untrusted data (e.g. check a signature first).
 */
 
-QByteArray doubleToData(double num)
-{
-    union
-    {
-        char tab[8];
-        double n;
-    } castUnion;
-    //char n[8];
-    //*((double*) n) = num;
-
-    castUnion.n=num;
-    return QByteArray(castUnion.tab,8);
-}
-
-QByteArray floatToData(float num)
-{
-    union
-    {
-        char tab[4];
-        float n;
-    } castUnion;
-
-    castUnion.n=num;
-    return QByteArray(castUnion.tab,4);
-}
-
-float dataToFloat(QByteArray data)
-{
-    union
-    {
-        char tab[4];
-        float n;
-    } castUnion;
-
-    castUnion.tab[0]=data.data()[0];
-    castUnion.tab[1]=data.data()[1];
-    castUnion.tab[2]=data.data()[2];
-    castUnion.tab[3]=data.data()[3];
-    return castUnion.n;
-}
-
-// Converts a string into PNet string data
-QByteArray stringToData(QString str)
-{
-    QByteArray data(4,0);
-    // Write the size in a Uint of variable lenght (8-32 bits)
-    int i=0;
-    uint num1 = (uint)str.toUtf8().size();
-    while (num1 >= 0x80)
-    {
-        data[i] = (unsigned char)(num1 | 0x80); i++;
-        num1 = num1 >> 7;
-    }
-    data[i]=num1;
-    data.resize(i+1);
-    data+=str.toUtf8();
-    return data;
-}
-
 QString dataToString(QByteArray data)
 {
-    // Variable UInt32
-    unsigned char num3;
-    int num = 0;
+    char num3;
+    int strlen = 0;
     int num2 = 0;
     int i=0;
     do
     {
-        num3 = data[i]; i++;
-        num |= (num3 & 0x7f) << num2;
+        num3 = data[i++];
+        strlen |= (num3 & 0x7f) << num2;
         num2 += 7;
     } while ((num3 & 0x80) != 0);
-    unsigned int strlen = (uint) num;
 
-    if (!strlen)
+    if (strlen <= 0)
         return QString();
 
-    data = data.right(data.size()-i); // Remove the strlen
+    // Remove the strlen
+    data.remove(0, i - 1);
```

Comment:
According to @tux3, this `data.remove(0, i - 1);` change has broken deserialization and by extension the auto-update feature in qTox since v1.5.0 (related issue https://github.com/qTox/qTox/issues/3910)